narwhals/_spark_like/expr.py

```diff
@@ -272,3 +288,153 @@ def var(self: Self, ddof: int) -> Self:
+            if closed == "both":
+                return (_input >= lower_bound) & (_input <= upper_bound)
+            if closed == "neither":
+                return (_input > lower_bound) & (_input < upper_bound)
+            if closed == "left":
+                return (_input >= lower_bound) & (_input < upper_bound)
+            return (_input > lower_bound) & (_input <= upper_bound)
+
+        return self._from_call(
+            _is_between,
+            "is_between",
+            lower_bound=lower_bound,
+            upper_bound=upper_bound,
+            returns_scalar=self._returns_scalar,
+        )
+
+    def is_duplicated(self) -> Self:
+        def _is_duplicated(_input: Column) -> Column:
+            from pyspark.sql import Window
+            from pyspark.sql import functions as F  # noqa: N812
+
+            return F.count(_input).over(Window.partitionBy(_input)) > 1
+
+        return self._from_call(
+            _is_duplicated, "is_duplicated", returns_scalar=self._returns_scalar
+        )
+
+    def is_finite(self) -> Self:
+        def _is_finite(_input: Column) -> Column:
+            from pyspark.sql import functions as F  # noqa: N812
+
+            # A value is finite if it's not NaN, not NULL, and not infinite
+            return (
+                ~F.isnan(_input)
+                & ~F.isnull(_input)
+                & (_input != float("inf"))
+                & (_input != float("-inf"))
+            )
+
+        return self._from_call(
+            _is_finite, "is_finite", returns_scalar=self._returns_scalar
+        )
+
+    def is_in(self, values: Sequence[Any]) -> Self:
+        def _is_in(_input: Column, values: Sequence[Any]) -> Column:
+            return _input.isin(values)
+
+        return self._from_call(
+            _is_in,
+            "is_in",
+            values=values,
+            returns_scalar=self._returns_scalar,
+        )
+
+    def is_nan(self) -> Self:
+        def _is_nan(_input: Column) -> Column:
+            from pyspark.sql import functions as F  # noqa: N812
+
+            # Need to handle both NaN and NULL values
+            return F.when(
+                F.isnan(_input) | F.isnull(_input),
+                F.lit(1),
+            ).otherwise(F.lit(0))
+
+        return self._from_call(_is_nan, "is_nan", returns_scalar=self._returns_scalar)
+
+    def is_unique(self) -> Self:
+        def _is_unique(_input: Column) -> Column:
+            from pyspark.sql import Window
+            from pyspark.sql import functions as F  # noqa: N812
+
+            return F.count(_input).over(Window.partitionBy(_input)) == 1
+
+        return self._from_call(
+            _is_unique, "is_unique", returns_scalar=self._returns_scalar
+        )
+
+    def len(self) -> Self:
+        def _len(_input: Column) -> Column:
+            from pyspark.sql import functions as F  # noqa: N812
+
+            return F.count(_input)
+
+        return self._from_call(_len, "len", returns_scalar=True)
```

Comment:
Unlike count, `len` should return the length including nulls (see [polars docs](https://docs.pola.rs/api/python/dev/reference/expressions/api/polars.Expr.len.html))